eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/inmemory/InMemoryEventStorageEngine.java

```diff
@@ -50,8 +50,19 @@
  */
 public class InMemoryEventStorageEngine implements EventStorageEngine {
 
-    private final NavigableMap<TrackingToken, TrackedEventMessage<?>> events = new ConcurrentSkipListMap<>();
-    private final Map<String, List<DomainEventMessage<?>>> snapshots = new ConcurrentHashMap<>();
+  protected final NavigableMap<TrackingToken, TrackedEventMessage<?>> events;
+  protected final Map<String, List<DomainEventMessage<?>>> snapshots;
+
+  public InMemoryEventStorageEngine() {
+    this(new ConcurrentSkipListMap<>(), new ConcurrentHashMap<>());
+  }
+
+  public InMemoryEventStorageEngine(
```

Comment:
This might be nitpicking since `InMemoryEventStorageEngine` is not intended for production use but I don't like that `events` and `snapshots` may be manipulated outside of this class. I'd propose getters for them which would return unmodifiable collections.

Reply:
Fair suggestion, @jangalinski might shed some light on how he intended to use this stuff.

Reply:
I want to run "minimal" axon apps that do not require anything to work. I will use these for training, testing and learning. I started by using an inMemoryEventStore that can print the list of events for demonstration purpose.
For this I need the maps "protected", so I can extend and access the fields. getters would be an option as well, good point with the read-only access to the maps from outside.

The other thing is partially related: In order to persist my eventStore, I started experimenting with MapDB, so I would use an InMemoryEventStore but exchange the maps-implementation.

The combination of protected-scope and no-arg/all-arg constructor is a solution for both these issues, but yes, it comes with the price that the events can be modified from the outside.
Maybe a quick fix would be to make the all-args constructor protected as well, so you only can controll the implementation via extension?

Reply:
Thank you for this explanation @jangalinski ! IMO the cleanest solution would be to introduce the `MapDBEventStorageEngine` instead of making `InMemoryEventStorageEngine` "durable". However, I see the benefit of re-using `InMemoryEventStorageEngine`. Making this constructor `protected` would communicate clearly what is its intention.

Reply:
Instead of making the fields `protected`, consider making them accessible by using a (`protected` or `public`) accessor method.